### PR TITLE
Archive list: account for ini overrides

### PR DESCRIPTION
**Pull request by SlavicPotato**
Any plugin may override sResourceArchiveList and sResourceArchiveList2 from which base archive lists are read by the game, hence this.

It scans priority order and selects the first instance of either option.

This will probably be obsoleted by a future Mutagen archives implementation. Currently it provides no way to do it easily.

There's also a concurrency fix in the bsa parallel loop.

### CODE REVIEW

File: ImmersiveEquipmentMeshGen/MeshHandler.cs [hunk 1]
```diff
@@ -11,6 +11,10 @@
 using nifly;
 using Mutagen.Bethesda.Plugins;
 using Mutagen.Bethesda.Archives;
+using IniParser;
```
Comment (SteveTownsend) [inline ImmersiveEquipmentMeshGen/MeshHandler.cs:14]: Won't this new dependency break user runs? Not sure how Nuget dependency management works outside of Visual Studio tbh. Seems like _IniParser_ would need to get rolled into the patcher download now.

File: ImmersiveEquipmentMeshGen/MeshHandler.cs [hunk 2]
```diff
@@ -300,6 +306,132 @@ public void GenerateMesh(NifFile nif, string originalPath, string newPath, Model
             }
         }
 
+        /* engine reverts to defaults if it can't read files (the archaic GetPrivateProfile api is used), we might as well 
+         * throw and force the user to sort out their stuff */
```
Comment (SteveTownsend) [inline ImmersiveEquipmentMeshGen/MeshHandler.cs:310]: comment out of sync with code?

**Review by SteveTownsend — COMMENTED** [review]
sorry for delay, I have Covid right now

**Comment by SlavicPotato** [comment]
Hope it's not too bad, get better!

File: ImmersiveEquipmentMeshGen/MeshHandler.cs (hunk 2 above)
Comment (SlavicPotato) [reply ImmersiveEquipmentMeshGen/MeshHandler.cs:310]: It's refering to OpenRead which throws if the file can't be read for any reason

File: ImmersiveEquipmentMeshGen/MeshHandler.cs (hunk 1 above)
Comment (SlavicPotato) [reply ImmersiveEquipmentMeshGen/MeshHandler.cs:14]: Mutagen uses it too (https://github.com/Mutagen-Modding/Mutagen/blob/b840d5248236559666072f0e71681c713b9a181b/Mutagen.Bethesda.Core/Archives/DI/GetArchiveIniListings.cs#L34), I'm guessing it shouldn't be an issue?

**Review by SteveTownsend — APPROVED** [review]

**Comment by SteveTownsend** [comment]
> Hope it's not too bad, get better!

thanks for the good wishes and fixing up the patcher ;-)

**Pull request merged**